drivers/sensor/stm32_vref/stm32_vref.c

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2023 Kenneth J. Miller <ken@miller.ec>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/drivers/adc.h>
+#include <zephyr/logging/log.h>
+#if defined(CONFIG_SOC_SERIES_STM32H5X)
+#include <stm32_ll_icache.h>
+#endif /* CONFIG_SOC_SERIES_STM32H5X */
+
+LOG_MODULE_REGISTER(stm32_vref, CONFIG_SENSOR_LOG_LEVEL);
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32_vref)
+#define DT_DRV_COMPAT st_stm32_vref
+#else
+#error "No compatible devicetree node found"
+#endif
```

Comment:
Shouldn't be required. The file won't be built when `CONFIG_STM32_VREF` is not enabled and it depends on `DT_HAS_ST_STM32_VREF_ENABLED`.